### DISCUSSION TITLE
Detect error page from _app

### DISCUSSION THREAD

**Discussion by alicerocheman** [vercel/next.js#28289] [Ideas]
### Describe the feature you'd like to request

I would like to detect from inside my _app component when I'm on an error page (no matter which)

I have not found this info anywhere in the _app component props or next router

### Describe the solution you'd like

I would like to see  `{ isErrorPage:boolean, errorCode: number }` in the router

### Describe alternatives you've considered
I haven't found any alternative

**Comment by icyJoseph** [comment]
Hi,


Perhaps you could check `useRouter().pathname` 

- if you have implemented `/pages/404.tsx` then `useRouter().pathname` will be `/404`
- Similarly, if you have `/pages/500.tsx`, then `useRouter().pathname` will be `/500`
- Otherwise it'll be `useRouter().pathname` will be `/_error`

You could also check `pageProps` in `_app` in case you were passing some data from the error to the component.

For example, with the `built-in` error page: 

```js
function MyApp({ Component, pageProps }: AppProps) {
  console.log(useRouter().pathname, pageProps); //  /_error, {statusCode: 500} 
  return <Component {...pageProps} />;
}
```

The log happens when I got to a page that uses `GSSP` and that just throws.
